code/game/objects/items/explosives/mine.dm

```diff
@@ -168,38 +227,56 @@
 
 
 /obj/item/explosive/mine/proc/set_tripwire()
-	if(!active && !tripwire)
+	if(!active && !tripwire && use_tripwire)
 		var/tripwire_loc = get_turf(get_step(loc, dir))
 		tripwire = new(tripwire_loc)
 		tripwire.linked_claymore = src
-		active = TRUE
+	active = TRUE
+
 
 
 //Mine can also be triggered if you "cross right in front of it" (same tile)
-/obj/item/explosive/mine/Crossed(atom/A)
+/obj/item/explosive/mine/Crossed(atom/victim)
 	..()
-	if(isliving(A))
-		var/mob/living/L = A
-		if(!L.stat == DEAD)//so dragged corpses don't trigger mines.
+	if(isliving(victim))
+		var/mob/living/unfortunate_soul = victim
+		if(heavy_trigger && buried)
+			unfortunate_soul.visible_message(SPAN_DANGER("[icon2html(src, viewers(src))] [name] clicks as [unfortunate_soul] moves in front of it."), \
+			SPAN_DANGER("[icon2html(src, unfortunate_soul)] [name] clicks as you move in front of it."), \
+			SPAN_DANGER("You hear a click."))
+			playsound(loc, 'sound/weapons/flipblade.ogg', 35, 1)
+
+		if(!unfortunate_soul.stat == DEAD)//so dragged corpses don't trigger mines.
 			return
 		else
-			try_to_prime(A)
+			try_to_prime(victim)
```

Comment:
this doesn't make sense. It triggers (if heavy_trigger and buried), but doesn't set off the mine if the corpse is dead? Have it either one way (no trigger) or the other (triggers and goes off)

Reply:
deadchecking moved to the try_to_prime proc